src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -156,23 +156,48 @@ public ExchangeService(BidderCatalog bidderCatalog,
 
         final long startTime = clock.millis();
 
-        return extractBidderRequests(bidRequest, uidsCookie, aliases, timeout)
+        // sanity check: discard imps without extension
+        final List<Imp> imps = imps(bidRequest);
+
+        final Map<String, List<String>> deprecatedBiddersErrors = new HashMap<>();
+
+        // identify valid biddersAndErrors and aliases out of imps
+        final List<String> bidders = biddersAndErrors(imps, deprecatedBiddersErrors, aliases);
+
+        return extractBidderRequests(bidRequest, imps, bidders, uidsCookie, aliases, timeout)
                 .map(bidderRequests ->
                         updateRequestMetric(bidderRequests, uidsCookie, aliases, publisherId, metricsContext))
                 .compose(bidderRequests -> CompositeFuture.join(bidderRequests.stream()
                         .map(bidderRequest -> requestBids(bidderRequest, startTime,
                                 auctionTimeout(timeout, cacheInfo.doCaching), aliases, bidAdjustments(requestExt),
                                 currencyRates(targeting)))
                         .collect(Collectors.toList())))
-                // send all the requests to the bidders and gathers results
+                // send all the requests to the biddersAndErrors and gathers results
```

Comment:
Check replacing though all other code, was `bidders` not `biddersAndErrors`

Reply:
checked before commit, missed this one(